### PR TITLE
feat: editable header

### PR DESCRIPTION
**Pull request by agurtovoy**
https://github.com/user-attachments/assets/faf441c2-5522-4b59-a625-3867a4833619

### CODE REVIEW

File: core/components/header/index.tsx [hunk 1]
```diff
@@ -5,18 +5,18 @@ import { getFormatter, getLocale, getTranslations } from 'next-intl/server';
 
 import { SearchResult } from '@/vibes/soul/primitives/navigation/index';
 import { localeSchema, searchSchema } from '@/vibes/soul/primitives/navigation/schema';
-import { HeaderSection } from '@/vibes/soul/sections/header-section';
 import { LayoutQuery } from '~/app/[locale]/(default)/query';
 import { getSessionCustomerAccessToken } from '~/auth';
 import { client } from '~/client';
 import { readFragment } from '~/client/graphql';
+import { getSearchResults } from '~/client/queries/get-search-results';
```
Comment (agurtovoy) [inline core/components/header/index.tsx:12]: Some imports got auto-rearranged here.

File: core/components/header/index.tsx [hunk 2]
```diff
@@ -5,18 +5,18 @@ import { getFormatter, getLocale, getTranslations } from 'next-intl/server';
 
 import { SearchResult } from '@/vibes/soul/primitives/navigation/index';
 import { localeSchema, searchSchema } from '@/vibes/soul/primitives/navigation/schema';
-import { HeaderSection } from '@/vibes/soul/sections/header-section';
```
Comment (agurtovoy) [inline core/components/header/index.tsx:8]: Replaced with `import { Header as HeaderSection } from '~/makeswift/components/header';`  below

File: core/components/header/index.tsx [hunk 3]
```diff
@@ -171,7 +171,10 @@ export const Header = async () => {
         logo: data.settings ? logoTransformer(data.settings) : undefined,
         searchAction: search,
         searchParamName: SEARCH_PARAM_NAME,
+        searchInputPlaceholder: t('searchPlaceholder'),
         searchCtaLabel: t('viewAll'),
+        emptySearchTitle: t('emptySearchTitle'),
+        emptySearchSubtitle: t('emptySearchSubtitle'),
```
Comment (agurtovoy) [inline core/components/header/index.tsx:177]: Localization for search messages.

File: core/vibes/soul/primitives/navigation/index.tsx [hunk 4]
```diff
@@ -72,7 +72,7 @@ interface Props {
   links: Link[];
   locales?: Locale[];
   activeLocaleId?: string;
-  localeAction: LocaleAction;
+  localeAction?: LocaleAction;
```
Comment (agurtovoy) [inline core/vibes/soul/primitives/navigation/index.tsx:75]: See VIBES PR here: https://github.com/makeswift/vibes/pull/296

**Review by migueloller — DISMISSED** [review]
LGTM

**Pull request merged**